clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -178,6 +178,23 @@ public CommitRequestManager(
      */
     @Override
     public NetworkClientDelegate.PollResult poll(final long currentTimeMs) {
+        if (coordinatorRequestManager.coordinator().isEmpty() && closing) {
+            CommitFailedException exception = new CommitFailedException(
+                    "Failed to commit offsets: Coordinator unknown and consumer is closing");
+
+            List<NetworkClientDelegate.UnsentRequest> pendingCommits = pendingRequests.drainPendingCommits();
```

Comment:
We could inline this variable.
`pendingRequests.drainPendingCommits().forEach...`